lib/puppet/provider/package/portage.rb

```diff
@@ -117,7 +117,7 @@ def qatom
       if match
         result_fields.zip(match.captures) do |field, value|
           # some fields can be empty or (null) (if we are not passed a category in the package name for instance)
-          if value == '(null)'
+          if value == '(null)' || value == '<unset>'
```

Comment:
If you use `[xxxx]` (as you do below), you will never see `(null)` or `<unset>` with any version of portage-utils.  Hence you should IMO drop this somewhat ugly bit, and just do the check at line 122 alone.